api/models.py

```diff
@@ -148,6 +153,24 @@ def create_indexes(cls, collection):
         collection.create_index("profile.username", unique=True)
 
 
+class TestUser(BeanieBaseUser, Document):
+    """Test user"""
+    username: Indexed(str, unique=True)
+
+
+class UserRead(schemas.BaseUser[PydanticObjectId]):
+    username: Indexed(str, unique=True)
+
+
+class UserCreate(schemas.BaseUserCreate):
+    username: Indexed(str, unique=True)
+
+
+class UserUpdate(schemas.BaseUserUpdate):
+    username: Optional[Indexed(str, unique=True)]
```

Comment:
Out of interest, why is this one optional whereas it's not in the other models?

Reply:
That's to allow users to only send fields that are updating. If I remove the `Optional`, the endpoint would expect `username` field in the request even if the user wants it to be unchanged.
If you look at the definition of the parent class `BaseUserUpdate`, you'll see all the fields are optional.
```
class BaseUserUpdate(CreateUpdateDictModel):
    password: Optional[str]
    email: Optional[EmailStr]
    is_active: Optional[bool]
    is_superuser: Optional[bool]
    is_verified: Optional[bool]
```

Reply:
Right, I think this brings us back to the idea I mentioned a while ago about having a different model for the API and for the database.